cp3pt0-deployment/setup_tenant.sh

```diff
@@ -156,33 +169,57 @@ function pre_req() {
     fi
 }
 
+function check_ns_list(){
+    for ns in $OPERATOR_NS $SERVICES_NS ${TETHERED_NS//,/ }; do
+        check_namespace $ns
+        if [ $? -ne 0 ]; then
+            error "Namespace $ns does not exist or current user $user does not get permission for this namespace\n"
+        fi
+    done
+}
+
 function create_ns_list() {
     for ns in $OPERATOR_NS $SERVICES_NS ${TETHERED_NS//,/ }; do
         create_namespace $ns
+        if [ $? -ne 0 ]; then
+            error "Namespace $ns cannot be created, please ensure user $user has proper permission to create namepace\n"
+        fi
     done
 }
 
 function setup_topology() {
+    if $LIMITED;then 
+        check_ns_list
```

Comment:
Wondering whether if it is necessary to have `check_ns_list`?

In `create_ns_list`, it already checks whether the namespace exists or not.

If it is not there, it will try to create namespace, and return error if user does not have permission to do so.
If it is there, script will move forward.